src/tests/test_certstor_system.cpp

```diff
@@ -76,6 +76,33 @@ Test::Result find_cert_by_subject_dn(Botan::Certificate_Store& certstore)
    return result;
    }
 
+Test::Result find_cert_by_utf8_subject_dn(Botan::Certificate_Store& certstore)
+   {
+   Test::Result result("System Certificate Store - Find Certificate by UTF8 subject DN");
+
+   try
+      {
+      auto dn = get_utf8_dn();
+
+      result.start_timer();
+      auto cert = certstore.find_cert(dn, std::vector<uint8_t>());
+      result.end_timer();
+
+      if(result.test_not_null("found certificate", cert.get()))
+         {
+         auto cns = cert->subject_dn().get_attribute("CN");
+         result.test_is_eq("exactly one CN", cns.size(), size_t(1));
+         result.test_eq("CN", cns.front(), "D-TRUST Root CA 3 2013");
```

Comment:
Fails on Ubuntu "Failure 1: System Certificate Store - Find Certificate by UTF8 subject DN CN produced unexpected result 'D-TRUST Root Class 3 CA 2 2009' expected 'D-TRUST Root CA 3 2013'"

Reply:
That's really strange ... Apparently the D-TRUST Root CA 3 2013 cert is not available on the Ubuntu agents, but it really shouldn't find a certificate with another name (and sadly it's also not yet reproducible for me ...)
I will investigate further.

Reply:
I added a test for the flatfile certstore, which tries to find the same DN and should succeed if it doesn't find it (as this failing test does find a cert, but the wrong one) and that works.
On the Linux agents, the system certstore is just a flatfile_cerstor with `/etc/ssl/certs/ca-certificates.crt` as the path, right? Could you maybe extract this file from the agents? I'd like to use the flatfile certstore test with the default system CA list and find out what might be going on (tried it with the list of Ubuntu 18.10, but that also worked as expected)

Reply:
Weird. I don't have any way to directly access the build VMs (that I know of) but for debugging purposes you can add arbitrary shell commands at the start of https://github.com/randombit/botan/blob/master/src/scripts/ci/setup_travis.sh and see the output in the build log.

Reply:
Indeed, with the CA list of the linux agent it's reproducible.